### PR TITLE
Add CAN beep

### PR DESCRIPTION
**Pull request by CUAVcaijie**
Add CAN buzzer

### CODE REVIEW

File: src/drivers/uavcan/sensors/beep.cpp [hunk 1]
```diff
@@ -0,0 +1,106 @@
+/****************************************************************************
+ *
+ *   Copyright (c) 2014, 2015 PX4 Development Team. All rights reserved.
```
Comment (TSC21) [inline src/drivers/uavcan/sensors/beep.cpp:3]:
```suggestion
 *   Copyright (c) 2020 PX4 Development Team. All rights reserved.
```

File: src/drivers/uavcan/sensors/beep.cpp [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+	_beep_pub(node),
+	_timer(node)
+{
+}
+
+int
+UavcanUavcanBeep::init()
+{
+	/*
+	 * Setup timer and call back function for periodic updates
```
Comment (TSC21) [inline src/drivers/uavcan/sensors/beep.cpp:53]: Can you move this description to the header file?

File: src/drivers/uavcan/sensors/beep.hpp [hunk 3]
```diff
@@ -0,0 +1,91 @@
+/****************************************************************************
+*
+*   Copyright (c) 2014, 2015 PX4 Development Team. All rights reserved.
```
Comment (TSC21) [inline src/drivers/uavcan/sensors/beep.hpp:3]:
```suggestion
*   Copyright (c) 2020 PX4 Development Team. All rights reserved.
```

File: src/drivers/uavcan/sensors/beep.hpp [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+* AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+* LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+* ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+* POSSIBILITY OF SUCH DAMAGE.
+*
+****************************************************************************/
+
+/**
+ * @file beep.hpp
+ *
```
Comment (TSC21) [inline src/drivers/uavcan/sensors/beep.hpp:36]: Can you add a description here with `@brief`?

File: src/drivers/uavcan/sensors/beep.hpp [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+#include <uavcan/uavcan.hpp>
+#include <uavcan/equipment/indication/BeepCommand.hpp>
+#include <uORB/Subscription.hpp>
+#include <uORB/topics/tune_control.h>
+#include <lib/tunes/tunes.h>
+
+/**
+ * @brief The UavcanUavcanBeep class
+ */
+
```
Comment (TSC21) [inline src/drivers/uavcan/sensors/beep.hpp:52]:
```suggestion
```

File: src/drivers/uavcan/sensors/beep.hpp [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+	/*
+	* setup periodic updater
+	*/
+	int init();
+
+private:
+	/*
+	 * Max update rate to avoid exessive bus traffic
+	 */
+	static constexpr unsigned			MAX_RATE_HZ = 100;
```
Comment (TSC21) [inline src/drivers/uavcan/sensors/beep.hpp:67]:
```suggestion
	static constexpr unsigned MAX_RATE_HZ = 100;
```

File: src/drivers/uavcan/sensors/beep.hpp [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+
+	void periodic_update(const uavcan::TimerEvent &);
+
+	typedef uavcan::MethodBinder<UavcanUavcanBeep *, void (UavcanUavcanBeep::*)(const uavcan::TimerEvent &)>
+	TimerCbBinder;
+
+	uORB::Subscription _tune_control_sub{ORB_ID(tune_control)};
+
+	/*
+	 * libuavcan related things
```
Comment (TSC21) [inline src/drivers/uavcan/sensors/beep.hpp:77]: Can you replace with details on what each thing is doing instead?

File: src/drivers/uavcan/sensors/beep.hpp [hunk 8]
```diff
@@ -0,0 +1,91 @@
...
+	 */
+	uavcan::Publisher<uavcan::equipment::indication::BeepCommand> _beep_pub;
+
+	uavcan::TimerEventForwarder<TimerCbBinder>			_timer;
```
Comment (TSC21) [inline src/drivers/uavcan/sensors/beep.hpp:81]:
```suggestion
	uavcan::TimerEventForwarder<TimerCbBinder> _timer;
```

File: src/drivers/uavcan/CMakeLists.txt [hunk 9]
```diff
@@ -130,6 +130,7 @@ px4_add_module(
 		sensors/flow.cpp
 		sensors/gnss.cpp
 		sensors/mag.cpp
+		sensors/beep.cpp
```
Comment (TSC21) [inline src/drivers/uavcan/CMakeLists.txt:133]: Can you please move this above and make sure it is alphabetically ordered?

File: src/drivers/uavcan/sensors/beep.cpp (hunk 2 above)
Comment (CUAVcaijie) [reply src/drivers/uavcan/sensors/beep.cpp:53]: ok, I will modify

File: src/drivers/uavcan/sensors/beep.hpp (hunk 4 above)
Comment (CUAVcaijie) [reply src/drivers/uavcan/sensors/beep.hpp:36]: ok

File: src/drivers/uavcan/sensors/beep.hpp (hunk 7 above)
Comment (CUAVcaijie) [reply src/drivers/uavcan/sensors/beep.hpp:77]: Ok

File: src/drivers/uavcan/CMakeLists.txt (hunk 9 above)
Comment (CUAVcaijie) [reply src/drivers/uavcan/CMakeLists.txt:133]: ok

**Pull request closed without merging**